edxval/views.py

```diff
@@ -373,3 +378,221 @@ def put(self, request):
         EncodedVideo.objects.create(video=video, profile=profile, **encode_data)
 
         return Response(status=status.HTTP_200_OK)
+
+
+class TranscriptCredentialsView(APIView):
+    """
+    API View to fetch and create Transcript provider credentials.
+    """
+    authentication_classes = (JwtAuthentication, SessionAuthentication)
+    permission_classes = (ReadRestrictedDjangoModelPermissions,)
+    queryset = TranscriptCredentials.objects.all()
+
+    def get(self, request):
+        """
+        Retrieves the transcript credentials for a given organization and provider.
+
+        **Example requests**:
+
+            GET api/val/v0/videos/transcript-credentials/?
+            provider=3play
+            &org=org
+
+        **GET Parameters**:
+
+            The following query parameters are required to get the credentials:
+
+                * provider(str): transcript provider, which is either 3PlayMedia or Cielo24.
+
+                * org(str): organization whose credentials are to be fetch.
+
+        **Response Values**
+
+            For a successful request, the following values are returned along with 200 status:
+
+                * api_key(str): provider key
+
+                * api_secret_key(str): provider api secret key(only for 3PlayMedia)
+
+                * provider(str): transcript provider
+
+                * org(str): organization whose credentials are fetched.
+
+            For the error, 400 response code is returned with:
+
+                * message(str): error message
+        """
+        response = validate_request_params(request.query_params, ['org', 'provider'])
+        if response:
+            return response
+
+        org = request.query_params['org']
+        provider = request.query_params['provider']
+        try:
+            credentials = TranscriptCredentials.objects.get(
+                provider=provider, org=org
+            )
+            status_code = status.HTTP_200_OK
+            data = dict(
+                api_key=credentials.api_key,
+                api_secret_key=credentials.api_secret,
+                org=credentials.org,
+                provider=credentials.provider
+            )
+        except TranscriptCredentials.DoesNotExist:
+            status_code = status.HTTP_400_BAD_REQUEST
+            data = {'message': "Credentials not found for provider {provider} & organization {org}".format(
+                provider=provider,
+                org=org
+            )}
+
+        return Response(status=status_code, data=data)
+
+    def get_cielo_token_response(self, username, api_secure_key):
```

Comment:
I'm wondering if it would be cleaner to separate out provider-specific stuff like this (and the settings that go with it) into separate source files.  Just seems a little out of place to have Cielo-specific stuff here in the middle of views.py.  What do you think?

Reply:
Sure, it will make the view code more compact. I will update the PR.

Reply:
The settings should be left in Django settings. The reason being that Cielo API URL is different. Right now, we are using a sandboxing URL. In VEDA, the URL is defined in edx-video-worker [instance-config.yaml](https://github.com/edx/edx-video-worker/blob/master/instance_config.yaml#L53) but it is sandboxing URL in [edx-video-pipeline](https://github.com/edx/edx-video-pipeline/blob/master/instance_config.yaml#L56). That's why I am thinking that we can override Cielo URL settings at deployment and use actual API URL. Although I will make the code to use defaults if some settings aren't found.